futures-util/src/async_await/join.rs

```diff
@@ -47,3 +47,64 @@ macro_rules! join {
         )*)
     } }
 }
+
+/// Attempts to poll multiple futures simultaneously, returning a `Result`
+/// of a tuple of all of the results once complete.
+///
+/// `try_join!` is similar to `join!`, but completes immediately if any of
+/// the futures return an error.
+///
+/// This macro is only usable inside of async functions, closures, and blocks.
+///
+/// # Examples
+///
+/// ```
+/// #![feature(pin, async_await, await_macro, futures_api)]
+/// # futures::executor::block_on(async {
+/// use futures::{try_join, future};
+///
+/// let a = future::ready(Ok::<i32, i32>(1));
+/// let b = future::ready(Err::<u64, i32>(2));
+///
+/// assert_eq!(try_join!(a, b), Err(2));
+///
+/// let a = future::ready(Ok::<i32, i32>(1));
+/// let b = future::ready(Ok::<u64, i32>(2));
+///
+/// assert_eq!(try_join!(a, b), Ok((1, 2)));
```

Comment:
Maybe make it two examples [like we have for `map_ok`](https://github.com/rust-lang-nursery/futures-rs/blob/43bfbbe450220db6710706cadd5975b268bb4415/futures-util/src/try_future/mod.rs#L114-L137). Also I suggest to show the successful one first